modules/post/windows/gather/checkvm.rb

```diff
@@ -40,85 +40,159 @@ module supports detection of Hyper-V, VMWare, VirtualBox, Xen, QEMU,
+  def key_present?(vm_k)
+    @keys.each_value do |v|
+      return true if v.include?(vm_k)
+    end
+    false
+  end
+
+  def get_srval(key)
+    srvals = registry_enumkeys(key)
+    srvals = [] if srvals.nil?
+    srvals
+  end
+
+  # returns true if regval matches a regex
+  def regval_match?(key, val, rgx)
+    return true if get_regval_str(key, val) =~ rgx
+
+    false
+  end
+
+  # returns true if regval is eql to a string
+  def regval_eql?(key, val, str)
+    get_regval_str(key, val) == str
   end
 
   def get_regval_str(key, valname)
     ret = registry_getvaldata(key, valname)
-    if ret.kind_of(Array)
+    if ret.is_a?(Array)
       ret = ret.join
     end
     ret
   end
 
+  def parallels?
+    @system_bios_version = get_regval_str('HKLM\\HARDWARE\\DESCRIPTION\\System', 'SystemBiosVersion')
+
+    @video_bios_version = get_regval_str('HKLM\\HARDWARE\\DESCRIPTION\\System', 'VideoBiosVersion')
+
+    if @system_bios_version =~ /parallels/i || @video_bios_version =~ /parallels/i
+      return true
+    end
+
+    false
+  end
 
   def hyperv?
     physical_host = get_regval_str('HKLM\\SOFTWARE\\Microsoft\\Virtual Machine\\Guest\\Parameters', 'PhysicalHostNameFullyQualified')
+
     if physical_host
       report_note(
         host: session,
         type: 'host.physicalHost',
         data: { physicalHost: physical_host },
         update: :unique_data
       )
+
       print_good("This is a Hyper-V Virtual Machine running on physical host #{physical_host}")
       return true
     end
 
     sfmsvals = registry_enumkeys('HKLM\\SOFTWARE\\Microsoft')
     if sfmsvals
-      return true if sfmsvals.include?('Hyper-V')
-      return true if sfmsvals.include?('VirtualMachine')
+      %w[Hyper-V VirtualMachine].each do |vm|
+        return true if sfmsvals.include?(vm)
+      end
     end
 
-    return true if get_regval_str('HKLM\\HARDWARE\\DESCRIPTION\\System', 'SystemBiosVersion') =~ /vrtual/i
-
-    %w[HKLM\\HARDWARE\\ACPI\\FADT HKLM\\HARDWARE\\ACPI\\RSDT].each do |key|
-      srvvals = registry_enumkeys(key)
-      return true if srvvals && srvvals.include?('VRTUAL')
+    if @system_bios_version =~ /vrtual/i || @system_bios_version == 'Hyper-V'
+      return true
     end
 
-    %w[vmicexchange vmicheartbeat vmicshutdown vmicvss].each do |service|
-      return true if service_exists?(service)
-    end
+    keys = %w[HKLM\\HARDWARE\\ACPI\\FADT HKLM\\HARDWARE\\ACPI\\RSDT HKLM\\HARDWARE\\ACPI\\DSDT]
+
+    register_keys(keys)
 
-    key_path = 'HKLM\\HARDWARE\\DESCRIPTION\\System'
-    system_bios_version = get_regval_str(key_path, 'SystemBiosVersion')
-    return true if system_bios_version && system_bios_version.include?('Hyper-V')
+    return true if key_present?('VRTUAL')
 
-    key_path = 'HKLM\\HARDWARE\\DEVICEMAP\\Scsi\\Scsi Port 0\\Scsi Bus 0\\Target Id 0\\Logical Unit Id 0'
-    return true if get_regval_str(key_path, 'Identifier') =~ /Msft    Virtual Disk    1.0/i
+    hyperv_services = %w[vmicexchange vmicshutdown vmicvss]
```

Comment:
Keeping it as a list as `services_exist` expects a list. 
```suggestion
    hyperv_services = %w[vmicexchange]
```